test/integration/host_js_test.rb

```diff
@@ -315,6 +332,23 @@ class HostJSTest < IntegrationTestWithJavascript
       click_link 'Parameters'
       assert_equal find("#s2id_host_lookup_values_attributes_#{lookup_key.id}_value .select2-chosen").text, "true"
     end
+
+    test 'changing host group updates parameters tab' do
+      hostgroup1, hostgroup2 = FactoryGirl.create_pair(:hostgroup, :with_parameter)
+      host = FactoryGirl.create(:host, :hostgroup => hostgroup1)
+
+      visit edit_host_path(host)
+      assert page.has_link?('Parameters', :href => '#params')
+      click_link 'Parameters'
+      assert page.has_selector?("#inherited_parameters #name_#{hostgroup1.group_parameters.first.name}")
+
+      click_link 'Host'
+      select2(hostgroup2.name, :from => 'host_hostgroup_id')
+      wait_for_ajax
+
+      click_link 'Parameters'
+      assert page.has_selector?("#inherited_parameters #name_#{hostgroup2.group_parameters.first.name}")
```

Comment:
Maybe it's worth asserting that the parameter from hostgroup 1 is not available at this point?